src/derivatives/linalg/reductions.jl

```diff
@@ -5,6 +5,16 @@
 # basic sum #
 #-----------#
 
+function Base.sum(x::AbstractArray{<:TrackedReal{V,D}}; dims = nothing) where {V,D}
+    tp = tape(x)
+    if dims === nothing
+        return reduce(+, x)
+    else
+        out = track(sum(value(x), dims = dims), D, tp)
+        record!(tp, SpecialInstruction, sum, (x, dims), out)
+    end
+    return out
+end
 function Base.sum(x::TrackedArray{V,D}) where {V,D}
```

Comment:
Just curious and trying to understand what's going on: could one instead of adding the definition above just widen the signature here to
```suggestion
 function Base.sum(x::AbstractArray{<:TrackedReal{V,D}}; dims=:) where {V,D}
```
and forward the keyword argument? IIRC `TrackedArray <: AbstractArray{<:TrackedReal}` so this should not break the existing definition.

Reply:
thanks I had to change the implementation slightly but I incorporated your suggestion